src/wq/ql/notify.py

```diff
@@ -79,20 +79,12 @@ def retry_intervals(self) -> [int]:
     def run(self):
         data = self.task_input()
         svc = data.fiware_service
-        svc_path = data.fiware_service_path if svc else None    # (*)
+        svc_path = data.fiware_service_path
         try:
             with translator_for(svc) as trans:
                 trans.insert(data.payload, svc, svc_path)
         except Exception as e:
             self._handle_exception(svc, e)
-# NOTE. Service path.
```

Comment:
@c0c0n3 i have no memory of why it was like this... this anyhow seems to introduce a bug when service path is not null, and service is... because whatever data sent will be cast to a null servicePath

Reply:
well, clearly given all the mighty bombing we have now on the reporter tests, highlights how this choice was "core" in the implementation, despite leading to uncorrect support of service_paths :/

Reply:
oh dear, it looks like I've actually introduced a bug! the code was meant to be like for like but obviously it isn't! I must've missed a statement when I copy-pasted the code. It was meant to be exactly the same as it was in the reporter before implementing wq...

Reply:
well, at least that was my intention judging by the comment in the code...

Reply:
i am not sure this was introduced by you :) the note was saying that the logic was taken from the original notify, so i assume that's true, so the question is why it was introduced originally. of course, given that we basically always test against a tenant, we don't observe any problem. testing with tenant `None` the are problems, which are solved by removing, the cast to `None` of the path, but the other comes up...

Reply:
so, i confirm it was there:
https://github.com/orchestracities/ngsi-timeseries-api/blob/45aff7a8a94ed4ea67e3cb6ccb93fa06fd7c6b5f/src/reporter/reporter.py#L179

and that is was introduced with multitenancy support 3 years ago:
https://github.com/orchestracities/ngsi-timeseries-api/pull/65/files

Reply:
oh my! it looks like it;'s a catch 22... :-)